test/e2e/pv_reclaimpolicy.go

```diff
@@ -0,0 +1,231 @@
+/*
```

Comment:
I like the idea of e2e reclaimPolicy tests. Would you consider making this test more generic so that reclaimPolicy can be run for additional providers (GCE, AWS, etc) including NFS? I was thinking of adding the  name "vsphere" to the inner `KubeDescribe`, and then, perhaps, adding `Context` (or just using separate `It` 's for the delete and retain tests. 

Or, if you think this file will get too large then separate _e2e/reclaim-<provider>_ files can be created.